libibverbs/cmd_ioctl.h

```diff
@@ -57,6 +58,18 @@ static inline uint64_t ioctl_ptr_to_u64(const void *ptr)
 #endif
 }
 
+static inline void scrub_ptr_attr(void *ptr)
+{
+#if UINTPTR_MAX == UINT64_MAX
+	/* Do nothing */
+#else
+	RDMA_UAPI_PTR(void *, data) *scrub_data;
+
+	scrub_data = container_of(ptr, typeof(*scrub_data), data);
+	scrub_data->data_data_u64 = ioctl_ptr_to_u64(scrub_data->data);
```

Comment:
Did anyone ever test this? That can't work.

ccan container_of has a bug where it does not notice that void * is type incompatible.

scrub_ptr_attr needs to take a void ** as argument then this code will be right.